slack-lint-checks/src/main/java/slack/lint/rx/RxObservableEmitDetector.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.rx
+
+import com.android.tools.lint.checks.DataFlowAnalyzer
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULambdaExpression
+import slack.lint.util.sourceImplementation
+
+class RxObservableEmitDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) {
+        val issue = functionToIssue[node.methodName] ?: return
+        val lambdaExpression =
+          requireNotNull(node.valueArguments.lastOrNull() as? ULambdaExpression) {
+            "Missing trailing lambda!"
+          }
```

Comment:
Lint's one area where we don't want to do offensive (as opposed to defensive) coding like this. Since this runs in the IDE, the user may not have written this argument yet. Instead, we should try it and just return early if it's not found

Reply:
Hmm ok, I hadn't considered that. I initially did just that until I realized that it improperly passed unit tests. I'll update and remove